pkg/auth/data/authconfig_data.go

```diff
@@ -94,7 +97,9 @@ func addAuthConfigCore(name, aType string, enabled, sloSupported bool, managemen
 	}
 	annotations[auth.CleanupAnnotation] = auth.CleanupRancherLocked
 
-	_, err := management.Management.AuthConfigs("").ObjectClient().Create(&v3.AuthConfig{
+	// technote: An ObjectClient is used for the creation to ensure that all fields of all the
+	// possible AuthConfig variants make it into the store unharmed.
```

Comment:
nit: I'm not really sure how helpful this comment is here as it doesn't really explains the why.
```suggestion
```

Reply:
I thought the why was in `ensure that all fields of all the possible AuthConfig variants make it into the store unharmed.`